test/integration/NodeRunner.spec.ts

```diff
@@ -0,0 +1,278 @@
+        "defaultSource": ["kraken"],
+        "interval": 0,
+        "priceAggregator": "median",
+        "sourceTimeout": 2000,
+        "tokens": {
+          "BTC": {
+           "source": ["coinbase"]
+          },
+          "ETH": {}
+        }
+      }`),
+      jwk,
+      nodeConfig
+    );
+
+    return expect(sut.run()).rejects.toThrowError("Could not determine maxPriceDeviationPercent");
+  });
+
+  it("should throw if no sourceTimeout", async () => {
+    //given
+    mockArProxy.getBalance.mockResolvedValue(0.2);
+    const sut = await NodeRunner.create(
+      JSON.parse(`{
+        "defaultSource": ["kraken"],
+        "interval": 0,
+        "priceAggregator": "median",
+        "maxPriceDeviationPercent": 25,
+        "tokens": {
+          "BTC": {
+           "source": ["coinbase"]
+          },
+          "ETH": {}
+        }
+      }`),
+      jwk,
+      nodeConfig
+    );
+
+    return expect(sut.run()).rejects.toThrowError("No timeout configured for");
+  });
+
+  it("should throw if Arweave balance too low on initial check", async () => {
+    //given
+    mockArProxy.getBalance.mockResolvedValue(0.1);
+    const sut = await NodeRunner.create(
+      manifest,
+      jwk,
+      nodeConfig
+    );
+
+    return expect(sut.run()).rejects.toThrowError("AR balance too low");
+  });
+
+
+  it("should broadcast fetched and signed prices", async () => {
+    //given
+    mockArProxy.getBalance.mockResolvedValue(0.2);
+
+    const sut = await NodeRunner.create(
+      manifest,
+      jwk,
+      nodeConfig
+    );
+
+    await sut.run();
+    expect(mockArProxy.prepareUploadTransaction).toHaveBeenCalledWith(
+      {
+        "Content-Encoding": "gzip",
+        "Content-Type": "application/json",
+        "app": "Redstone",
+        "timestamp": "111111111",
+        "type": "data",
+        "version": "0.4",
+      },
+      [{
+        "id": "00000000-0000-0000-0000-000000000000",
+        "source": {"coinbase": 444, "kraken": 445},
+        "symbol": "BTC",
+        "timestamp": 111111111,
+        "value": 444.5,
+        "version": "0.4"
+      }],
+    );
+    expect(mockArProxy.sign).toHaveBeenCalledWith(
+      "{\"id\":\"00000000-0000-0000-0000-000000000000\",\"permawebTx\":\"mockArTransactionId\",\"provider\":\"mockArAddress\",\"source\":{\"coinbase\":444,\"kraken\":445},\"symbol\":\"BTC\",\"timestamp\":111111111,\"value\":444.5,\"version\":\"0.4\"}"
+    );
+    expect(axios.post).toHaveBeenCalledWith(
+      "http://broadcast.test",
+      [
+        {
+          "id": "00000000-0000-0000-0000-000000000000",
+          "permawebTx": "mockArTransactionId",
+          "provider": "mockArAddress",
+          "signature": "mock_signed",
+          "source": {"coinbase": 444, "kraken": 445},
+          "symbol": "BTC",
+          "timestamp": 111111111,
+          "value": 444.5,
+          "version": "0.4"
+        }
+      ]
+    );
+    expect(mockArProxy.postTransaction).not.toHaveBeenCalled();
```

Comment:
Why don't we post AR transactions here?

Reply:
cause we're simulating test env in this test.
there is a separate test case "should save transaction on Arweave in mode=PROD"